flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/MultiJoinTestPrograms.java

```diff
@@ -0,0 +1,461 @@
+                                    .addSchema(
+                                            "user_id_0 STRING PRIMARY KEY NOT ENFORCED",
+                                            "name STRING",
+                                            "cash INT")
+                                    .addOption("changelog-mode", "I,UA,D")
+                                    .producedValues(
+                                            Row.ofKind(RowKind.INSERT, "1", "Gus", 100),
+                                            Row.ofKind(RowKind.INSERT, "2", "Bob", 20),
+                                            Row.ofKind(RowKind.INSERT, "3", "Nomad", 50),
+                                            Row.ofKind(RowKind.INSERT, "4", "David", 5),
+                                            Row.ofKind(RowKind.INSERT, "5", "Eve", 0),
+                                            Row.ofKind(RowKind.INSERT, "6", "Frank", 70),
+                                            Row.ofKind(RowKind.INSERT, "7", "Welcher", 100),
+                                            Row.ofKind(RowKind.INSERT, "8", "Joe no order", 10),
+                                            Row.ofKind(RowKind.INSERT, "9", "Charlie Smith", 50),
+                                            Row.ofKind(RowKind.DELETE, "2", "Bob", 20),
+                                            Row.ofKind(
+                                                    RowKind.UPDATE_AFTER,
+                                                    "9",
+                                                    "Charlie Taylor",
+                                                    50))
+                                    .build())
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("Orders")
+                                    .addSchema(
+                                            "user_id_1 STRING",
+                                            "order_id STRING PRIMARY KEY NOT ENFORCED",
+                                            "product STRING")
+                                    .addOption("changelog-mode", "I,D")
+                                    .producedValues(
+                                            Row.ofKind(RowKind.INSERT, "1", "order0", "ProdB"),
+                                            Row.ofKind(RowKind.INSERT, "1", "order1", "ProdA"),
+                                            Row.ofKind(RowKind.INSERT, "2", "order2", "ProdB"),
+                                            Row.ofKind(RowKind.INSERT, "3", "order3", "ProdC"),
+                                            Row.ofKind(RowKind.INSERT, "4", "order4", "ProdD"),
+                                            Row.ofKind(RowKind.INSERT, "6", "order6", "ProdF"),
+                                            Row.ofKind(RowKind.INSERT, "7", "order7", "ProdG"),
+                                            Row.ofKind(RowKind.INSERT, "9", "order9", "ProdA"))
```

Comment:
looks like the ids are mostly perfectly ordered and also always at column 0, I suggest to reorder some and move columns also to the middle or end.

Reply:
I've shuffled rows and the columns for the four way complex join for restore and semantic test. Also shuffled columns for the rel/exec/explain tests 👍 https://github.com/apache/flink/pull/26687/commits/570da7bb0ae69f416ef84bfd431dc4a853b3b061